src/lib/core.ts

```diff
@@ -1,44 +1,39 @@
 import { bytesToBigInt } from '../bindings/crypto/bigint-helpers.js';
 import { defineBinable } from '../bindings/lib/binable.js';
 import { sizeInBits } from '../provable/field-bigint.js';
-import { Bool, Field, Scalar, Group } from '../snarky.js';
+import { Bool, Scalar, Group } from '../snarky.js';
+import { Field as InternalField } from './field.js';
 import { Scalar as ScalarBigint } from '../provable/curve-bigint.js';
 import { mod } from '../bindings/crypto/finite_field.js';
 
 export { Field, Bool, Scalar, Group };
 
-type InternalConstantField = { value: [0, Uint8Array] };
+/**
+ * An element of a finite field.
+ */
+const Field = toFunctionConstructor(InternalField);
```

Comment:
Here is where `Field` is turned from a class into a "class that can also be used as a function", to support `Field(x)`